packages/api/src/external/commonwell/command/cw-patient-data/update-cw-data.ts

```diff
@@ -45,19 +47,30 @@ export async function updateCwPatientData({
       lock: true,
     });
 
-    return updateCwPatientDataWithinDBTx(cwPatientData, existingPatient, transaction);
+    return updateCwPatientDataWithinDBTx(
+      cwPatientData,
+      existingPatient,
+      transaction,
+      invalidateLinks
+    );
   });
   return updateResult.dataValues;
 }
 
 export async function updateCwPatientDataWithinDBTx(
   update: CwPatientDataUpdate,
   existing: CwPatientDataModel,
-  transaction: Transaction
+  transaction: Transaction,
+  linksToInvalidate?: CwLink[]
 ): Promise<CwPatientDataModel> {
   const { data: newData } = update;
   const updatedLinks = [...(newData.links ?? []), ...existing.data.links];
-  const uniqueUpdatedLinks = uniqBy(updatedLinks, function (nl) {
+
+  const validLinks = linksToInvalidate
+    ? updatedLinks.filter(link => isContainedAt(link, linksToInvalidate))
```

Comment:
Shouldn't this return updated links _not_ contained in the list of links to invalidate?

Reply:
Good catch